server/src/api/comment.rs

```diff
@@ -282,29 +282,64 @@ impl Perform for Oper<EditComment> {
       if user.banned {
         return Err(APIError::err("site_ban").into());
       }
+    } else {
+      // check that user can mark as read
+      let parent_id = orig_comment.parent_id;
+      match parent_id {
+        Some(pid) => {
+          let parent_comment =
+            blocking(pool, move |conn| CommentView::read(&conn, pid, None)).await??;
+          if !(user_id == parent_comment.creator_id) {
+            return Err(APIError::err("no_comment_edit_allowed").into());
+          }
+        }
+        None => {
+          let parent_post = blocking(pool, move |conn| Post::read(conn, edit_id)).await??;
```

Comment:
This needs to be `orig_comment.post_id`, the `edit_id` is the comment id.

Reply:
good catch, copy-paste fail on my part.